os/board/common/partitions.c

```diff
@@ -0,0 +1,141 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <debug.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+
+#include <tinyara/fs/mtd.h>
+#include <tinyara/fs/ioctl.h>
+
+void configure_partitions(void)
+{
+#if defined(CONFIG_FLASH_PART)
+	int partno;
+	int partoffset;
+	const char *parts = CONFIG_FLASH_PART_LIST;
+	const char *types = CONFIG_FLASH_PART_TYPE;
+#if defined(CONFIG_MTD_PARTITION_NAMES)
+	const char *names = CONFIG_FLASH_PART_NAME;
+#endif
+	FAR struct mtd_dev_s *mtd;
+	FAR struct mtd_geometry_s geo;
+
+	mtd = progmem_initialize();
```

Comment:
Progmem is to write data on flash directly instead of ram.
So, to use this, each board should implement functions in progmem.
Currently only artik05x supports it, you have to call up_flashinitialize for other boards.

Reply:
Thanks. I fixed it.